lighthouse-core/gather/gatherers/full-page-screenshot.js

```diff
@@ -92,18 +92,23 @@ class FullPageScreenshot extends Gatherer {
 
       return nodes;
     }
-    const expression = `(function () {
-      ${pageFunctions.getBoundingClientRectString};
-      return (${resolveNodes.toString()}());
-    })()`;
+
+    /**
+     * @param {boolean} useIsolation
+     */
+    function evaluateInPage(useIsolation) {
```

Comment:
Adding another evaluate function seems unnecessary. I think it's better to just use `passContext.driver.evaluate` twice.

Reply:
> Adding another evaluate function seems unnecessary. I think it's better to just use `passContext.driver.evaluate` twice.

Bump, WDYT?

Reply:
I normally wouldn't DRY up so aggressively (just two usages here...), but given the unfortunately "boilerplate"ness here, and the fact that it is all local (as in, this is a nest function only used immediately after being defined) it seems ok to me.

my goal was more readability, which I realized could be further improved by not using a boolean param / using the same function name, so I refactored a bit.